[executorch] Always generate CustomOpsNativeFunctions.h if custom_ops.yaml is present

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95084

To match the build system logic, enforce CustomOpsNativeFunctions.h to be generated if we have custom_ops.yaml, even if we don't select any custom ops.

Added unit test.

Differential Revision: [D43402718](https://our.internmc.facebook.com/intern/diff/D43402718)

larryliu0820 committed:
[executorch] Always generate CustomOpsNativeFunctions.h if custom_ops.yaml is present

To match the build system logic, enforce CustomOpsNativeFunctions.h to be generated if we have custom_ops.yaml, even if we don't select any custom ops.

Added unit test.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[executorch] Always generate CustomOpsNativeFunctions.h if custom_ops.yaml is present"


To match the build system logic, enforce CustomOpsNativeFunctions.h to be generated if we have custom_ops.yaml, even if we don't select any custom ops.

Added unit test.

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43402718).

iseeyuan commented on torchgen/gen_executorch.py:
```diff
@@ -291,15 +291,28 @@ def gen_functions_declarations(
 def gen_headers(
     *,
     native_functions: Sequence[NativeFunction],
+    gen_custom_ops_header: bool,
```
What's the overhead if we always generate CustomOpsNativeFunctions.h?

larryliu0820 replied:
Yeah I didn’t want to generate it if there’s no custom ops. Let me land this to unblock internal use case and then follow up.

iseeyuan reviewed (approved):
LGTM. If we could simplify to always generate CustomOpsNativeFunctions.h without much overhead it's even better.

larryliu0820 commented:
@pytorchbot merge

larryliu0820 commented:
@pytorchbot rebase

larryliu0820 committed:
Update on "[executorch] Always generate CustomOpsNativeFunctions.h if custom_ops.yaml is present"


To match the build system logic, enforce CustomOpsNativeFunctions.h to be generated if we have custom_ops.yaml, even if we don't select any custom ops.

Added unit test.

Differential Revision: [D43402718](https://our.internmc.facebook.com/intern/diff/D43402718)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43402718).

larryliu0820 commented:
@pytorchbot merge
